common/mails/courses/index.ts

```diff
@@ -230,4 +253,21 @@ export async function sendPupilCourseSuggestion(course: Course | Prisma.course,
             courseURL: `https://app.lern-fair.de/single-course/${subcourse.id}`,
         });
     }
+
+    if (isCourseValid(publishedAt, courseCapacity, alreadyPromoted)) {
+        for (let pupil of pupils) {
+            if (!courseSubject) {
+                notify(pupil);
+            } else {
+                const subjects = parseSubjectString(pupil.subjects);
+                const isPupilsSubject = subjects.some((subject) => subject.name == courseSubject);
```

Comment:
I would move this to line 259 to then check the `if (!courseSubject || isPupilsSubject) { notify(pupil) }`